pkg/combustion/registry_test.go

```diff
@@ -0,0 +1,155 @@
+			ContainerImages: []image.ContainerImage{
+				{
+					Name: "hello-world:latest",
+				},
+				{
+					Name:           "rgcrprod.azurecr.us/longhornio/longhorn-ui:v1.5.1",
+					SupplyChainKey: "carbide-key.pub",
+				},
+			},
+			HelmCharts: []image.HelmChart{
+				{
+					Name:    "rancher",
+					RepoURL: "https://releases.rancher.com/server-charts/stable",
+					Version: "2.8.0",
+				},
+			},
+		},
+	}
+
+	// Test
+	err := writeHaulerManifest(ctx)
+
+	// Verify
+	require.NoError(t, err)
+
+	manifestFileName := filepath.Join(ctx.BuildDir, haulerManifestYamlName)
+	_, err = os.Stat(manifestFileName)
+	require.NoError(t, err)
+
+	foundBytes, err := os.ReadFile(manifestFileName)
+	require.NoError(t, err)
+	found := string(foundBytes)
+	assert.Contains(t, found, "rgcrprod.azurecr.us/longhornio/longhorn-ui:v1.5.1")
+	assert.Contains(t, found, "https://releases.rancher.com/server-charts/stable")
+}
+
+func TestWriteHaulerManifestEmptyManifest(t *testing.T) {
+	// Setup
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	// Test
+	err := writeHaulerManifest(ctx)
+
+	// Verify
+	require.NoError(t, err)
+
+	manifestFileName := filepath.Join(ctx.BuildDir, haulerManifestYamlName)
+	_, err = os.Stat(manifestFileName)
+	require.NoError(t, err)
+
+	foundBytes, err := os.ReadFile(manifestFileName)
+	require.NoError(t, err)
+	found := string(foundBytes)
+	assert.Contains(t, found, "Embedded-Registry-Images")
+	assert.Contains(t, found, "Embedded-Registry-Charts")
+}
+
+func TestCreateRegistryCommand(t *testing.T) {
+	// Setup
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	// Test
+	cmd, logFile, err := createRegistryCommand(ctx, "testName", []string{"--flag", "test"})
+
+	// Verify
+	require.NoError(t, err)
+	require.NotNil(t, cmd)
+
+	expectedCommand := "testName"
+	expectedArgs := []string{"testName", "--flag", "test"}
+
+	assert.Equal(t, expectedCommand, cmd.Path)
+	assert.Equal(t, expectedArgs, cmd.Args)
+
+	assert.Equal(t, logFile, cmd.Stdout)
+	assert.Equal(t, logFile, cmd.Stderr)
+
+	foundFile := filepath.Join(ctx.BuildDir, "embedded-registry.log")
+	_, err = os.ReadFile(foundFile)
+	require.NoError(t, err)
+}
+
+func TestWriteRegistryScript(t *testing.T) {
+	// Setup
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	// Test
+	_, err := writeRegistryScript(ctx)
+
+	// Verify
+	require.NoError(t, err)
+
+	registryScriptPath := filepath.Join(ctx.CombustionDir, "13-embedded-registry.sh")
+	_, err = os.Stat(registryScriptPath)
+	require.NoError(t, err)
+
+	foundBytes, err := os.ReadFile(registryScriptPath)
+	require.NoError(t, err)
+	found := string(foundBytes)
+	assert.Contains(t, found, registryTarName)
+}
+
+func TestCopyHaulerBinary(t *testing.T) {
+	// Setup
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	haulerExecutable := filepath.Join(ctx.BuildDir, "hauler")
+	err := os.WriteFile(haulerExecutable, []byte(""), fileio.ExecutablePerms)
+	assert.NoError(t, err)
```

Comment:
We can't continue if this fails.
```suggestion
	require.NoError(t, err)
```